infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -32,20 +32,23 @@
     static final int BATCH_SIZE = 50;
     private StreamLog streamLog;
     private BlockingQueue<BatchWriterOperation> operationsQueue;
-    final ExecutorService writerService = Executors
-            .newSingleThreadExecutor(new ThreadFactoryBuilder()
-                    .setDaemon(false)
-                    .setNameFormat("LogUnit-Write-Processor-%d")
-                    .build());
+    private final ExecutorService writerService;
 
     /**
      * Returns a new BatchWriter for a stream log.
      *
+     * @param context   A server context
      * @param streamLog stream log for writes (can be in memory or file)
      */
-    public BatchWriter(StreamLog streamLog) {
+    public BatchWriter(@Nonnull ServerContext context,
+                       @Nonnull StreamLog streamLog) {
         this.streamLog = streamLog;
         operationsQueue = new LinkedBlockingQueue<>();
+        writerService = Executors
```

Comment:
We should have them all in one place, probably not a properties file though, probably serverContext is fine. It also makes shutdown a lot easier.

I didn't move them all in this PR because I thought it'd be a little much.